tests/src/test/scala/scalaz/UnapplyTest.scala

```diff
@@ -22,6 +22,102 @@ object UnapplyTest extends SpecLite {
     Unapply2[Arrow, Kleisli[NonEmptyList, Int, String]].TC: Arrow[Kleisli[NonEmptyList, ?, ?]]
   }
 
+  object unapply3 {
+    implicit def functorInstance[A]: Functor[λ[α => A]] = ???
+    implicitly[Unapply[Functor, String]]
```

Comment:
This can be written without the `implicitly[...]` wrapping, because `object Unapply` has an appropriate `apply`; see e.g. [`val O` in README](https://github.com/scalaz/scalaz/tree/v7.3.0-M16#standalone-type-class-usage).

Also, [testing `M` might be valuable](https://github.com/TomasMikula/scalaz/compare/Unapply...S11001001:Unapply).

Reply:
Thanks, I added your commit to this PR.